### PR TITLE
Clio 39037 support callouts hipchat

### PR DESCRIPTION
**Pull request by jamescolesnash**
@achengscode Could you please review this.

It needs two environment variables set for your machine.

Examples:
HUBOT_GOOGLE_CALENDAR_KEYFILE=file_path
HUBOT_GOOGLE_CALENDAR_EMAIL=support-ride-along-cliobot@support-ride-along-cliobot.iam.gserviceaccount.com

Keyfile can be found here:
https://console.developers.google.com/iam-admin/serviceaccounts/project?project=support-ride-along-cliobot

Currently it is set up to have access to my account, but you can just share your calendar with support-ride-along-cliobot@support-ride-along-cliobot.iam.gserviceaccount.com to see your items.

### CODE REVIEW

File: scripts/google_calendar.coffee [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        email: process.env.HUBOT_GOOGLE_CALENDAR_EMAIL
+        keyFile: process.env.HUBOT_GOOGLE_CALENDAR_KEYFILE
+        scopes: ['https://www.googleapis.com/auth/calendar.readonly']
+    _.extend(params, opt)
+
+    robot.logger.debug(params)
+
+    requestWithJWT(params, (err, res, body) ->
+      if err
+        console.log err
```
Comment (achengscode) [inline scripts/google_calendar.coffee]: I'm not sure if console.log is the best way to show the error, but at the same time, I'm not sure how to display errors anyway for a chat bot. Maybe just a simple "Sorry, calendar not found" reply?

File: scripts/google_calendar.coffee [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        for i, calendarItem of calendarListResponse.items
+          strs.push calendarItem.id
+        str strs
+    )
+
+  getCalendarEvents = (baseDate, startTime, calendarId, str) ->
+    request(
+      { url: "https://www.googleapis.com/calendar/v3/calendars/#{calendarId}" }
+      (calendarResponse) ->
+        timeMin = new Date(baseDate.getTime())
```
Comment (achengscode) [inline scripts/google_calendar.coffee]: You're using `moment` below; can you use `moment` here as well?

File: scripts/google_calendar.coffee [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    if event.end
+      strs.push "~"
+      if event.end.date
+        strs.push event.end.date
+      else if event.end.dateTime
+        strs.push event.end.dateTime.match justTime
+
+    strs.push event.summary
+    strs.push "\n"
+    strs.push "     #{event.htmlLink}"
```
Comment (achengscode) [inline scripts/google_calendar.coffee]: Whats with the extra spaces here?

If you want a tab, I think "\t" might do the trick.

File: scripts/google_calendar.coffee [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+  request = (opt, onSuccess) ->
+    params =
+      jwt:
+        email: process.env.HUBOT_GOOGLE_CALENDAR_EMAIL
+        keyFile: process.env.HUBOT_GOOGLE_CALENDAR_KEYFILE
+        scopes: ['https://www.googleapis.com/auth/calendar.readonly']
+    _.extend(params, opt)
+
+    robot.logger.debug(params)
```
Comment (achengscode) [inline scripts/google_calendar.coffee]: Don't know if you need the logger.debug here still, once it's deployed.

File: scripts/google_calendar.coffee (hunk 3 above)
Comment (jamescolesnash) [inline scripts/google_calendar.coffee]: Thanks, works.

File: scripts/google_calendar.coffee (hunk 4 above)
Comment (jamescolesnash) [inline scripts/google_calendar.coffee]: Removed

File: scripts/google_calendar.coffee (hunk 2 above)
Comment (jamescolesnash) [inline scripts/google_calendar.coffee]: Doesn't seem to work, 'moment.date' is just a number, and i need to get the time and date out of the 'Date'

File: scripts/google_calendar.coffee (hunk 1 above)
Comment (jamescolesnash) [inline scripts/google_calendar.coffee]: Fixed.

**Comment by jamescolesnash** [comment]
@achengscode  back to you.

**Comment by rbroemeling** [comment]
Closing without merge since the base issue (CLIO-39037) has been closed as WONTDO.

**Pull request closed without merging**